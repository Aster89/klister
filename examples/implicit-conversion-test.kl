#lang "implicit-conversion.kl"

(example
  (let-implicit string-length
    (+ 1 "foo")))
-- =>
-- (example
--   (let-implicit string-length
--     (#%app +
--            (#%integer-literal 1)
--            (#%string-literal "foo"))))
-- =>
-- (example
--   (let-implicit string-length
--     (my-app my-+
--             (my-integer-literal 1)
--             (my-string-literal "foo"))))
-- =>
-- (example
--   (run-reader
--     ((pick-reader-ap-variant) ((pick-reader-ap-variant) (reader-pure +)
--                                                         (reader-pure 1))
--                               (reader-pure "foo"))
--     string-length)
-- =>
-- (example
--   (run-reader
--     (reader-ap-with (lambda (str2int) str2int)
--                     (reader-ap (reader-pure +)
--                                (reader-pure 1))
--                     (reader-pure "foo"))
--     string-length))
-- =>
-- (example
--   (+ 1 (string-length "foo")))
-- =>
-- (example
--   (+ 1 3))
-- =>
-- (example
--   4)
