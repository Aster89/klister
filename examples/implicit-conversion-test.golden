42 : Integer
4 : Integer
